### PR TITLE
Impliment new header layout and visual style

### PR DESCRIPTION
**Pull request by frechg**
screen shot 2015-01-25 at 8 25 03 pm

### CODE REVIEW

**Comment by croaky** [comment]
Code looks good. Tests are failing because email_address is nil in some cases. Might need to define email_address method on Guest returning "". Should also check the production database to see if there are any User records with nil email addresses. User.where(email_address: nil)

**Comment by croaky** [comment]
Do we need the "sign out" link showing at all times? Current version is nice and clean with it hidden behind a click of the username. If we kept just the sign out hidden, it might also give us more room across the header for users with longer names.

**Comment by salbertson** [comment]
What about putting the signout link somewhere else, like a simple footer?

**Comment by croaky** [comment]
I had the same thought but my list of repos is crazy long. It also conceptually makes sense to me to group authentication with my name/avatar.

**Comment by frechg** [comment]
Yeah, sign out link could move to a footer. I am just progressively adding the new UI right now and adding a footer will come later if it still makes sense.

**Comment by frechg** [comment]
@croaky what do you think about only showing the avatar right now if the current_user has an email?

File: app/views/shared/_header.haml [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-        .menu
-          = link_to 'Your Repos', repos_path, class: 'repo-link'
-          = link_to "Your Account", account_path
-          = link_to 'Configure Hound', configuration_path
-          = link_to 'FAQ', faq_path
-          = link_to 'Sign Out', sign_out_path
+    %ul.pull-right
+      %li
+        = link_to account_path, class: "account" do
+          - unless current_user.email_address.nil?
```
Comment (graysonwright) [inline app/views/shared/_header.haml]: I think we should try to avoid a double-negative here. How about `- if current_user.email_address`

Comment (salbertson) [inline app/views/shared/_header.haml]: You could do something like this to support blank email addresses, not only nil ones:

```
- if current_user.email_address.present?
...
```

**Comment by salbertson** [comment]
One comment, looks good to me. :+1:

File: app/views/shared/_header.haml [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
-          = link_to "Your Account", account_path
-          = link_to 'Configure Hound', configuration_path
-          = link_to 'FAQ', faq_path
-          = link_to 'Sign Out', sign_out_path
+    %ul.pull-right
+      %li
+        = link_to account_path, class: "account" do
+          - if current_user.email_address.present?
+            .avatar(style="background-image: url('#{avatar_url(current_user)}')")
+          %span #{current_user.github_username}
```
Comment (gylaz) [inline app/views/shared/_header.haml]: How about `%span = current_user.github_username` instead for consistency.

**Comment by croaky** [comment]
> @croaky what do you think about only showing the avatar right now if the current_user has an email?

Works for me.

**Pull request closed without merging**